numba/_lapack.c

```diff
@@ -0,0 +1,1035 @@
+            PyGILState_Release(st);
+        }
+        return -1;
+    }
+    if (raw_func == NULL)
+        return -1;
+
+    _m = (F_INT) m;
+    _n = (F_INT) n;
+    _lda = (F_INT) lda;
+
+    (*(xxgetrf_t) raw_func)(&_m, &_n, a, &_lda, ipiv, info);
+    return 0;
+}
+
+
+/* Compute the inverse of a matrix given its LU decomposition*/
+NUMBA_EXPORT_FUNC(F_INT)
+numba_xxgetri(char kind, F_INT n, void *a, F_INT lda, F_INT *ipiv, \
+              void *work, F_INT *lwork, F_INT *info)
+{
+    void *raw_func = NULL;
+    F_INT _n, _lda;
+
+    switch (kind)
+    {
+        case 's':
+            raw_func = get_clapack_sgetri();
+            break;
+        case 'd':
+            raw_func = get_clapack_dgetri();
+            break;
+        case 'c':
+            raw_func = get_clapack_cgetri();
+            break;
+        case 'z':
+            raw_func = get_clapack_zgetri();
+            break;
+        default:
+        {
+            PyGILState_STATE st = PyGILState_Ensure();
+            PyErr_SetString(PyExc_ValueError,\
+                            "invalid kind of *inversion from LU factorization \
+            function");
+            PyGILState_Release(st);
+        }
+        return -1;
+    }
+    if (raw_func == NULL)
+        return -1;
+
+    _n = (F_INT) n;
+    _lda = (F_INT) lda;
+
+    (*(xxgetri_t) raw_func)(&_n, a, &_lda, ipiv, work, lwork, info);
+    return 0;
+}
+
+/* Compute the Cholesky factorization of a matrix */
+NUMBA_EXPORT_FUNC(int)
+numba_xxpotrf(char kind, char uplo, Py_ssize_t n, void *a, Py_ssize_t lda, int
+              *info)
+{
+    void *raw_func = NULL;
+    int _n, _lda;
+
+    switch (kind)
+    {
+        case 's':
+            raw_func = get_clapack_spotrf();
+            break;
+        case 'd':
+            raw_func = get_clapack_dpotrf();
+            break;
+        case 'c':
+            raw_func = get_clapack_cpotrf();
+            break;
+        case 'z':
+            raw_func = get_clapack_zpotrf();
+            break;
+        default:
+        {
+            PyGILState_STATE st = PyGILState_Ensure();
+            PyErr_SetString(PyExc_ValueError,
+                            "invalid kind of Cholesky factorization function");
+            PyGILState_Release(st);
+        }
+        return -1;
+    }
+    if (raw_func == NULL)
+        return -1;
+
+    _n = (int) n;
+    _lda = (int) lda;
+
+    (*(xxpotrf_t) raw_func)(&uplo, &_n, a, &_lda, info);
+    return 0;
+}
+
+
+/*
+ * cast_from_X()
+ * cast from a kind (s, d, c, z) = (float, double, complex, double complex)
+ * to a Fortran integer.
+ *
+ * Parameters:
+ * kind the kind of val
+ * val  a pointer to the value to cast
+ *
+ * Returns:
+ * A Fortran int from a cast of val (in complex case, takes the real part).
+ *
+ */
+// a template would be nice
+// struct access via non c99 (python only) cmplx types, used for compatibility
+F_INT cast_from_X(char kind, void *val)
```

Comment:
Why not simply `static`? There's a macro in `_numba_common.h`, but I don't see the point here.